src/common/util.cpp

```diff
@@ -465,10 +465,23 @@ std::string get_nix_version_display_string()
       pathRet = "/";
     else
       pathRet = pszHome;
-#ifdef MAC_OSX
+#ifdef __APPLE__
+    #include "TargetConditionals.h"
+    #ifdef TARGET_OS_MAC
     // Mac
-    pathRet /= "Library/Application Support";
-    config_folder =  (pathRet + "/" + CRYPTONOTE_NAME);
+      if(access((pathRet + "/." + CRYPTONOTE_NAME).c_str(), 0) == -1)
+      {
+        pathRet += "/Library/Application Support";
+        config_folder = (pathRet + "/" + CRYPTONOTE_NAME);
+      }
+      else
+      {
+        config_folder = (pathRet + "/." + CRYPTONOTE_NAME);
+        MWARNING("Current output directory '" << config_folder << "' should be moved to '" << (pathRet + "/Library/Application Support/" + CRYPTONOTE_NAME)  << "'.");
```

Comment:
Should there really be a warning? ~/.bitmonero is a valid location and is currently the default. This PR would cause every mac user to start getting a warning for no valid reason.

Reply:
I wouldn't mind. Just added it per first comment.
>There should be some kind of check for this, even if just an error on ~/.monerod existing with a message asking to move/remove it.

Reply:
We could just log it. Which level would you apply on it?

Reply:
My idea was that if ~/.bitmonero exists, you'd get a warning, since it means you'd have existing data where it wouldn't be used. But given the above comment ("plenty of apps use the former style of a .folder in ~"), I'm not sure it needs fixing in the first place.

This is a question for Mac users.

Reply:
Well I'm gonna talk for myself but I'd rather have Monero in the "Application Support" folder. 
As a programmer I have all "hidden" files (those starting with a dot) displayed because I need it. However this brings a mess in the User's directory when apps use that location as implied by the issue OP. The Application Support directory enhances the global UX and prevents accidental unwanted user actions as it places the Monero folder in a hidden folder itself (rather than the visible ~/).

Reply:
@moneromooo-monero Well I am a Mac user and the default has always been `~/.bitmonero` and hence any change here is going to cause issues. Also, to put it into context how many apps use ~/.xyz, on my laptop there are 153 different items here! It is very much normal Mac practice.

There is only one case I can think of to allow using `~/Library/Application Support/Monero` and that is if (and I believe it is, but I don't use it) the Monero GUI defaults to this location. All reasons I believe we should just silently allow both locations (not just checking the folder exists but checking it has data in it), and further, in my view, default to ~/.bitmonero simply because that is more of the standard for command line apps on Mac.

Reply:
I think the best reason to **not** change the path is actually to allow an easier assistance should a user have a problem. Their folder path would be the same regardless of the OS. 

Considering this we could remove the entire definition of the config_folder as we would use the default Unix path.

Reply:
@jtgrassie 
> if (and I believe it is, but I don't use it) the Monero GUI defaults to this location.

No, the GUI also uses the same `~/.bitmonero` folder, of course.